test/stablex/batch_exchange.js

```diff
@@ -55,6 +55,15 @@ contract("BatchExchange", async accounts => {
 
     BATCH_TIME = (await batchExchange.BATCH_TIME.call()).toNumber()
   })
+
+  // In the following tests, it might be possible that an batchId is read from the blockchain
+  // and in the next moment this batchId is no longer the current one. In order to prevent these
+  // situations, we set the adjust the start-time of each test to the start of an new auction.
+  beforeEach(async () => {
+    const batchExchange = await BatchExchange.deployed()
```

Comment:
Interesting trick, that you are using the globally deployed instance to advance the time. This relies on the fact that all instances (even the ones created in each test) use the same interval for defining batches.

Reply:
Not sure I understand how this is the case. In my mind I assume before Each runs "before each test" and interpret this as follows

```
// Coming from before each
const batchExchange = await BatchExchange.deployed()
// As seen inside each test
const contract = await BatchExchange.new()
```

How could the deployed version catch the new contract that hasn't been deployed yet?

Reply:
> How could the deployed version catch the new contract that hasn't been deployed yet?

It does not need to, as all instances will calculate the batch times the same way.

Reply:
batch0 always start on 1.1.1970 regardless of when the instance was deployed. Thus advancing time in one instance advances the batchId for all other (even not yet deployed instances)